rhr_flex_model:

  # Publish all joint states -----------------------------------

  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50  


  # Position Controllers ---------------------------------------

  # D parameter causes a lot of jitter, don't need it.
  # I doesn't seem to do anything.
  # Really only need a small P.

  preshape_1_position_controller:
    type: effort_controllers/JointPositionController
    joint: preshape_1
    pid: {p: 3.0, i: 0.0, d: 0.0}

  proximal_joint_1_position_controller:
    type: effort_controllers/JointPositionController
    joint: proximal_joint_1
    pid: {p: 3.0, i: 0.0, d: 0.0}

  distal_joint_1_position_controller:
    type: effort_controllers/JointPositionController
    joint: distal_joint_1
    pid: {p: 3.0, i: 0.0, d: 0.0}


  preshape_2_position_controller:
    type: effort_controllers/JointPositionController
    joint: preshape_2
    pid: {p: 3.0, i: 0.0, d: 0.0}

  proximal_joint_2_position_controller:
    type: effort_controllers/JointPositionController
    joint: proximal_joint_2
    pid: {p: 3.0, i: 0.0, d: 0.0}

  distal_joint_2_position_controller:
    type: effort_controllers/JointPositionController
    joint: distal_joint_2
    pid: {p: 3.0, i: 0.0, d: 0.0}


  proximal_joint_3_position_controller:
    type: effort_controllers/JointPositionController
    joint: proximal_joint_3
    pid: {p: 3.0, i: 0.0, d: 0.0}

  distal_joint_3_position_controller:
    type: effort_controllers/JointPositionController
    joint: distal_joint_3
    pid: {p: 3.0, i: 0.0, d: 0.0}

